### PR TITLE
Add description about `tidb_auth_token` authentication

### PR DESCRIPTION
**Pull request by CbcWestwolf**
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/pull/38585

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -193,7 +193,8 @@ mysql> SELECT * FROM t1;
 - 作用域：GLOBAL
 - 是否持久化到集群：是
 - 默认值：`mysql_native_password`
-- 可选值：`mysql_native_password`，`caching_sha2_password`，`tidb_sm3_password`
+- 可选值：`mysql_native_password`，`caching_sha2_password`，`tidb_sm3_password`，`tidb_auth_token`
+- `tidb_auth_token` 为 TiDB Cloud 内部实现，**不推荐设置为该值**。
```
Comment (xhebox) [inline system-variables.md:197]:
```suggestion
- `tidb_auth_token` 为 TiDB Cloud 内部所用，**不要设置为该值**。
```

Comment (xhebox) [reply system-variables.md:197]: Ditto for other two occurences.

**Review by xhebox — APPROVED** [review]

File: security-compatibility-with-mysql.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ aliases: ['/docs-cn/dev/security-compatibility-with-mysql/','/docs-cn/dev/refere
 
 TiDB 支持多种身份验证方式。通过使用 [`CREATE USER`](/sql-statements/sql-statement-create-user.md) 语句和 [`ALTER USER`](/sql-statements/sql-statement-create-user.md) 语句，即可创建新用户或更改 TiDB 权限系统内的已有用户。TiDB 身份验证方式与 MySQL 兼容，其名称与 MySQL 保持一致。
 
-TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务器和客户端建立连接时，如要指定服务器对外通告的默认验证方式，可通过 [`default_authentication_plugin`](/system-variables.md#default_authentication_plugin) 变量进行设置。`tidb_sm3_password` 为仅在 TiDB 支持的 SM3 身份验证方式，使用该方式登录的用户需要使用 [TiDB-JDBC](https://github.com/pingcap/mysql-connector-j/tree/release/8.0-sm3)。
+TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务器和客户端建立连接时，如要指定服务器对外通告的默认验证方式，可通过 [`default_authentication_plugin`](/system-variables.md#default_authentication_plugin) 变量进行设置。`tidb_sm3_password` 为仅在 TiDB 支持的 SM3 身份验证方式，使用该方式登录的用户需要使用 [TiDB-JDBC](https://github.com/pingcap/mysql-connector-j/tree/release/8.0-sm3)。`tidb_auth_token` 为仅用于 TiDB Cloud 内部的基于 JWT 的认证方式。
```
Comment (Oreoxmt) [inline security-compatibility-with-mysql.md:19]:
```suggestion
TiDB 目前支持的身份验证方式可在以下的表格中查找到。服务器和客户端建立连接时，如要指定服务器对外通告的默认验证方式，可通过 [`default_authentication_plugin`](/system-variables.md#default_authentication_plugin) 变量进行设置。`tidb_sm3_password` 为仅在 TiDB 支持的 SM3 身份验证方式，使用该方式登录的用户需要使用 [TiDB-JDBC](https://github.com/pingcap/mysql-connector-j/tree/release/8.0-sm3)。`tidb_auth_token` 为仅用于 TiDB Cloud 内部的基于 JSON Web Token (JWT) 的认证方式。
```

File: system-variables.md [hunk 3]
```diff
@@ -193,7 +193,8 @@ mysql> SELECT * FROM t1;
 - 作用域：GLOBAL
 - 是否持久化到集群：是
 - 默认值：`mysql_native_password`
-- 可选值：`mysql_native_password`，`caching_sha2_password`，`tidb_sm3_password`
+- 可选值：`mysql_native_password`，`caching_sha2_password`，`tidb_sm3_password`，`tidb_auth_token`
+- `tidb_auth_token` 为 TiDB Cloud 内部所用，**不要设置为该值**。
```
Comment (Oreoxmt) [inline system-variables.md:197]:
```suggestion
- `tidb_auth_token` 认证方式仅用于 TiDB Cloud 内部实现，**不要设置为该值**。
```

**Review by Oreoxmt — APPROVED** [review]

File: tidb-configuration-file.md [hunk 4]
```diff
@@ -323,6 +323,24 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 默认值：""，支持 TLSv1.1 及以上版本。
 + 可选值：`"TLSv1.0"`、`"TLSv1.1"`、`"TLSv1.2"` 和 `"TLSv1.3"`
 
+### `auth-token-jwks` <span class="version-mark">从 v6.4.0 版本开始引入</span>
+
+> **警告：**
+>
+> `tidb_auth_token` 认证方式仅用于 TiDB Cloud 内部实现，**不要修改该配置**。
+
++ 设置用于 `tidb_auth_token` 认证方式的 JWKS 的本地文件地址。
```
Comment (Oreoxmt) [inline tidb-configuration-file.md:332]:
```suggestion
+ 设置用于 `tidb_auth_token` 认证方式的 JSON Web Key Sets (JWKS) 的本地文件地址。
```

File: tidb-configuration-file.md [hunk 5]
```diff
@@ -323,6 +323,24 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
++ 默认值：""
+
+### `auth-token-refresh-interval` <span class="version-mark">从 v6.4.0 版本开始引入</span>
+
+> **警告：**
+>
+> `tidb_auth_token` 认证方式仅用于 TiDB Cloud 内部实现，**不要修改该配置**。
+
++ 设置用于 `tidb_auth_token` 认证方式的 JWKS 刷新时间间隔。
```
Comment (Oreoxmt) [inline tidb-configuration-file.md:341]:
```suggestion
+ 设置用于 `tidb_auth_token` 认证方式的 JSON Web Key Sets (JWKS) 刷新时间间隔。
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

File: tidb-configuration-file.md [hunk 6]
```diff
@@ -323,6 +323,24 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
++ 设置用于 `tidb_auth_token` 认证方式的 JSON Web Key Sets (JWKS) 的本地文件地址。
++ 默认值：""
+
+### `auth-token-refresh-interval` <span class="version-mark">从 v6.4.0 版本开始引入</span>
+
+> **警告：**
+>
+> `tidb_auth_token` 认证方式仅用于 TiDB Cloud 内部实现，**不要修改该配置**。
+
++ 设置用于 `tidb_auth_token` 认证方式的 JSON Web Key Sets (JWKS) 刷新时间间隔。
```
Comment (shichun-0415) [inline tidb-configuration-file.md:341]:
```suggestion
+ 设置 `tidb_auth_token` 认证方式的 JWKS 刷新时间间隔。
```

File: tidb-configuration-file.md [hunk 7]
```diff
@@ -323,6 +323,24 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 默认值：""，支持 TLSv1.1 及以上版本。
 + 可选值：`"TLSv1.0"`、`"TLSv1.1"`、`"TLSv1.2"` 和 `"TLSv1.3"`
 
+### `auth-token-jwks` <span class="version-mark">从 v6.4.0 版本开始引入</span>
+
+> **警告：**
+>
+> `tidb_auth_token` 认证方式仅用于 TiDB Cloud 内部实现，**不要修改该配置**。
+
++ 设置用于 `tidb_auth_token` 认证方式的 JSON Web Key Sets (JWKS) 的本地文件地址。
```
Comment (shichun-0415) [inline tidb-configuration-file.md:332]:
```suggestion
+ 设置 `tidb_auth_token` 认证方式的 JSON Web Key Sets (JWKS) 的本地文件地址。
```

**Review by shichun-0415 — APPROVED** [review]
Rest LGTM

**Comment by Oreoxmt** [comment]
DNM until https://github.com/pingcap/docs-cn/pull/11716 is merged

File: system-variables.md [hunk 8]
```diff
@@ -193,7 +193,8 @@ mysql> SELECT * FROM t1;
 - 作用域：GLOBAL
 - 是否持久化到集群：是
 - 默认值：`mysql_native_password`
-- 可选值：`mysql_native_password`，`caching_sha2_password`，`tidb_sm3_password`
+- 可选值：`mysql_native_password`，`caching_sha2_password`，`tidb_sm3_password`，`tidb_auth_token`
+- `tidb_auth_token` 认证方式仅用于 TiDB Cloud 内部实现，**不要设置为该值**。
 - 服务器和客户端建立连接时，这个变量用于设置服务器对外通告的默认身份验证方式。如要了解该变量的其他可选值，参见[可用的身份验证插件](/security-compatibility-with-mysql.md#可用的身份验证插件)。
```
Comment (qiancai) [inline system-variables.md:198]: 只是记录下：这个变化应该不需要体现到 TiDB v6.4.0 release notes 里，对用户看起来没有实际的作用

File: tidb-configuration-file.md [hunk 9]
```diff
@@ -323,6 +323,24 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 默认值：""，支持 TLSv1.1 及以上版本。
 + 可选值：`"TLSv1.0"`、`"TLSv1.1"`、`"TLSv1.2"` 和 `"TLSv1.3"`
```
Comment (qiancai) [inline tidb-configuration-file.md:325]: 只是记录下：这个变化应该不需要体现到 TiDB v6.4.0 release notes 里，对用户看起来没有实际的作用

**Pull request merged**